### PR TITLE
fix: corrected implementation of where param numpy mean

### PR DESCRIPTION
**Pull request by ShreyanshBardia**
# PR Description

## Related Issue

part of [task](https://trello.com/c/Y2fwIg7z/2698-implement-proper-handling-for-where-argument-in-the-numpy-frontend)

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by ShreyanshBardia** [comment]
I'll fix the failing test for `ndarray_mean` soon, after that this should be ready for final review

File: ivy/functional/frontends/numpy/ndarray/ndarray.py [hunk 1]
```diff
@@ -165,9 +165,9 @@ def any(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
     def argsort(self, *, axis=-1, kind=None, order=None):
         return np_frontend.argsort(self, axis=axis, kind=kind, order=order)
 
-    def mean(self, *, axis=None, dtype=None, out=None, keepdims=False, where=True):
+    def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
         return np_frontend.mean(
-            self,
+            self.ivy_array,
```
Comment (AnnaTz) [inline ivy/functional/frontends/numpy/ndarray/ndarray.py:170]: I don't think `.ivy_array` is needed here. `self` is a frontend array and frontend functions are designed to handle this type of input.

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py [hunk 2]
```diff
@@ -2766,42 +2769,51 @@ def test_numpy_ndarray_max(
...
     on_device,
 ):
-    input_dtypes, x, axis = dtype_x_axis
+    input_dtypes, x, axis = dtype_and_x
+    where, input_dtypes, test_flags = np_frontend_helpers.handle_where_and_array_bools(
+        where=where,
+        input_dtype=input_dtypes,
+        test_flags=method_flags,
+    )
+    if "bfloat16" in input_dtypes + dtype or []:
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2795]: Ideally we should filter `bfloat16` in `@handle_frontend_method`.

**Review by AnnaTz — CHANGES_REQUESTED** [review]
Very minor changes suggested.

File: ivy/functional/frontends/numpy/ndarray/ndarray.py (hunk 1 above)
Comment (ShreyanshBardia) [reply ivy/functional/frontends/numpy/ndarray/ndarray.py:170]: yes right, will update it

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py (hunk 2 above)
Comment (ShreyanshBardia) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_ndarray/test_ndarray.py:2795]: Earlier due to some issues, `bfloat16` was getting generated for `numpy` as well, but it looks like it has been resolved now. so will remove it

**Review by AnnaTz — APPROVED** [review]
Looks good, thanks!

**Pull request merged**